dashboard-api/internal/types/studyconfig.go

```diff
@@ -0,0 +1,125 @@
+package types
+
+import (
+	"context"
+	"encoding/json"
+	"reflect"
+	"time"
+)
+
+type DistributionVar string
+
+const (
+	Location DistributionVar = "location"
+	Gender   DistributionVar = "gender"
+	Age      DistributionVar = "age"
+)
+
+type StudyConfigRepository interface {
+	Create(ctx context.Context, dsc DatabaseStudyConf) error
+}
+
+// StudyConfig is the root config that holds the various
+// different types of study configuration that can be
+// applied to a study that gets sent by the frontend
+type StudyConfig struct {
```

Comment:
The database expects each config separately, which leads to an odd struct iteration when "transforming for database" - and the database version, `[]DatabaseStudyConf`, is an array and therefore doesn't even validate that they're all there. And one can imagine updating one without updating the others. 

To add to that, one can imagine that a user wants to "partially create" a study, create and save SOME of the confs, without saving all of them. 

Given all of that, I'm not sure it's right for the frontend to send `StudyConfig` as this large object, rather than just send each config individually. 

I think also if we think of validation as an first-class, but potentially asynchronous, process. It might be nice to have a `studies/:id/validate` endpoint that validates all the current configuration and returns any issues. As a simple MVP it miht just validate that it has all the confs.

Reply:
I have added an issue for the validation endpoint https://github.com/vlab-research/vlab/issues/122 and made the discussed changes for now.  I will work on the patch once this is done.